testng/src/main/java/cucumber/api/testng/TestNGCucumberRunner.java

```diff
@@ -68,6 +69,10 @@ public void runCucumber(CucumberFeature cucumberFeature) {
         }
     }
 
+    public void runScenario(PickleEvent pickle) throws Throwable {
+        runtime.runPickle(pickle);
```

Comment:
Not using a listener as in `runCucumber` make Ambiguous and Undefined result to be missed (TestNG seems to detect exceptions from the step definitions, but not the exceptions for Ambiguous and Undefined result which are thrown internally in the runner). I'll provide a Listener class (similar to the `FeatureResultListener`) which I had to make to get to the bottom of this behaviour.